sg_bookmarks.user.js

```diff
@@ -17,28 +17,59 @@
 // @grant        GM_listValues
 // @grant        GM_deleteValue
 // @grant        GM_getResourceText
+// @grant        GM_addStyle
 // ==/UserScript==
 
 /*jshint multistr: true */
 
 use(SgApi);
 use(Util);
 
+//Runtime variables
 var data = new Data();
+var lazyTrainManager = {};//Used to keep track of which giveaways are a train and identify the start and end of said trains.Format:{{desc1:String}: {timesOccured:int},{desc2:String}: {timesOccured:int},...}
+var prevNavRow;
+var navRowIsTrain = false;
+//In order to prevent racing issues, bookmarks are obtained using AJAX. However, this may cause a problem if the user navigates away from the page if the AJAX is not complete yet.
+//queuedBookmarkIds is there to catch these interrupted AJAX requests and ensure the integrity of the bookmarks. Before an AJAX call, the bookmark id is added to this map; after it is done,
+//the ID is removed. If the ID still exists in the map on navigating to a new page, the AJAX call is automatically retried.
+var queuedBookmarkIds = getQueuedBookmarkIds();
+
+//Settings
+var SETTINGS_STATES = "Show giveaway status";
+var SETTINGS_TRAIN = "Group giveaways in a train";
+var settings = new Settings("SG_Bookmarks")
```

Comment:
You shouldn't create a new Settings-object. Every Settings-object gets it's own section in /account/profile/. We want your Settings to be in the already existing "SG Bookmarks" sections. So you need extend the existing one (data.settings) in the method initSettings (Line 342) and then use that one.
Also, don't fiddle around too much with the settings menu. I'm sitting on a merge for the SgApi Settings Module that brings some new features and bugfixes (like description not being shown for some settings)